students/EricAdams/Iterators_and_Generators_Packaging/generator_lab/generator_solution.py

```diff
@@ -0,0 +1,74 @@
+"""
+generator_solution.py
+
+Solutions for the generator lab excercises.
+
+"""
+
+
+def intsum():
+    '''sum using generator
+    '''
+    x = list(range(0, 1000))
+    total = 0
+    for y in x:
+        total = total + y
+        yield total
+
+
+def doubler():
+    '''Double the previous value using a generator
+    '''
+    x = list(range(1, 1000))
+    for i, y in enumerate(x):
+        yield 2 ** i
+
+
+def fib(n):
+    '''fibonacci generator
+    Thanks to https://www.python-course.eu/generators.php
+    '''
+    a, b, counter = 1, 1, 0
+    while True:
+        if (counter > n):
+            return
+        yield a
+        a, b = b, a + b
+        counter += 1
+
+
+def prime():
+    """ Generate an infinite sequence of prime numbers.
+    This code is straight outta stackoverflow.
```

Comment:
was it a generator on SO? 

Frankly, I can't quite follow this (looking at it for a few minutes) can you?